packages/traverse/traverse.0.3.0/opam

```diff
@@ -16,6 +16,7 @@ depends: [
   "metapp" {>= "0.3.0"}
   "metaquot" {>= "0.3.0"}
   "odoc" {with-doc & >= "1.5.1"}
+  "ppxlib" {< "0.26.0"}
```

Comment:
It is not traverse that fails, but metapp:
```
#=== ERROR while compiling metapp.0.4.3 =======================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.13.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.13/.opam-switch/build/metapp.0.4.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p metapp -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/metapp-19-07576f.env
# output-file          ~/.opam/log/metapp-19-07576f.out
### output ###
# (cd _build/default && .ppx/e7c203cacf1cd3bb02e6242329effa31/ppx.exe --cookie 'library-name="metapp"' -o metapp/metapp.pp.mli --intf metapp/metapp.mli -corrected-suffix .ppx-corrected -diff-cmd - -dump-ast)
# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.13/lib/ocaml, /home/opam/.opam/4.13/lib/ocaml/compiler-libs
# File "_none_", line 1:
# Warning 58 [no-cmx-file]: no cmx file was found in path for module Metapp_api, and its interface was not compiled with -opaque
# (cd _build/default && .ppx/e7c203cacf1cd3bb02e6242329effa31/ppx.exe --cookie 'library-name="metapp"' -o metapp/metapp.pp.ml --impl metapp/metapp.ml -corrected-suffix .ppx-corrected -diff-cmd - -dump-ast)
# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.13/lib/ocaml, /home/opam/.opam/4.13/lib/ocaml/compiler-libs
# File "_none_", line 1:
# Warning 58 [no-cmx-file]: no cmx file was found in path for module Metapp_api, and its interface was not compiled with -opaque
# File "_none_", line 1:
# Warning 58 [no-cmx-file]: no cmx file was found in path for module Metapp_preutils, and its interface was not compiled with -opaque
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlopt.opt -open Stdcompat -w +32+34-40 -g -I metapp/.metapp.objs/byte -I metapp/.metapp.objs/native -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.13/lib/ocaml/compiler-libs -I /home/opam/.opam/4.13/lib/ppx_derivers -I /home/opam/.opam/4.13/lib/ppxlib -I /home/opam/.opam/4.13/lib/ppxlib/ast -I /home/opam/.opam/4.13/lib/ppxlib/astlib -I /home/opam/.opam/4.13/lib/ppxlib/print_diff -I /home/opam/.opam/4.13/lib/ppxlib/stdppx -I /home/opam/.opam/4.13/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.13/lib/sexplib0 -I /home/opam/.opam/4.13/lib/stdcompat -I /home/opam/.opam/4.13/lib/stdlib-shims -I preutils/.metapp_preutils.objs/byte -I preutils/.metapp_preutils.objs/native -intf-suffix .ml -no-alias-deps -o metapp/.metapp.objs/native/metapp.cmx -c -impl metapp/metapp.pp.ml)
# File "metapp/metapp.ml", line 323, characters 23-26:
# 323 |           if check_pat arg then
#                              ^^^
# Error: This expression has type
#          string Ppxlib__.Import.Ast.loc list * Ppxlib__.Import.Ast.pattern
#        but an expression was expected of type
#          Ppxlib.pattern = Astlib.Ast_414.Parsetree.pattern
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlc.opt -open Stdcompat -w +32+34-40 -g -bin-annot -I metapp/.metapp.objs/byte -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.13/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.13/lib/ocaml/compiler-libs -I /home/opam/.opam/4.13/lib/ppx_derivers -I /home/opam/.opam/4.13/lib/ppxlib -I /home/opam/.opam/4.13/lib/ppxlib/ast -I /home/opam/.opam/4.13/lib/ppxlib/astlib -I /home/opam/.opam/4.13/lib/ppxlib/print_diff -I /home/opam/.opam/4.13/lib/ppxlib/stdppx -I /home/opam/.opam/4.13/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.13/lib/sexplib0 -I /home/opam/.opam/4.13/lib/stdcompat -I /home/opam/.opam/4.13/lib/stdlib-shims -I preutils/.metapp_preutils.objs/byte -intf-suffix .ml -no-alias-deps -o metapp/.metapp.objs/byte/metapp.cmo -c -impl metapp/metapp.pp.ml)
# File "metapp/metapp.ml", line 323, characters 23-26:
# 323 |           if check_pat arg then
#                              ^^^
# Error: This expression has type
#          string Ppxlib__.Import.Ast.loc list * Ppxlib__.Import.Ast.pattern
#        but an expression was expected of type
#          Ppxlib.pattern = Astlib.Ast_414.Parsetree.pattern
```
This should not be needed

Reply:
Ooh, cool, that's probably the same for a couple of other packages as well. Thanks!

Reply:
Lookign at the logs, I think all of

- [metapp.0.4.0 (failed: This expression has type)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a8aa66bb463131a112f5b73f2430e04ec04d82f2/variant/opam-2.1,compilers,4.13,ppxlib.0.26.0,revdeps,metapp.0.4.0)
- [metapp.0.4.1 (failed: This expression has type)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a8aa66bb463131a112f5b73f2430e04ec04d82f2/variant/opam-2.1,compilers,4.13,ppxlib.0.26.0,revdeps,metapp.0.4.1)
- [metapp.0.4.2 (failed: This expression has type)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a8aa66bb463131a112f5b73f2430e04ec04d82f2/variant/opam-2.1,compilers,4.13,ppxlib.0.26.0,revdeps,metapp.0.4.2)
- [metapp.0.4.3 (failed: This expression has type)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/a8aa66bb463131a112f5b73f2430e04ec04d82f2/variant/opam-2.1,compilers,4.13,ppxlib.0.26.0,revdeps,metapp.0.4.3)

need the upper bound and will avoid a number of failures